wallet/sync.go

```diff
@@ -407,14 +407,103 @@ func (w *Wallet) scanAddressIndex(start int, end int, account uint32,
 // have been used by an HD keychain stemming from this wallet in the default
 // account.
 func (w *Wallet) rescanActiveAddresses() error {
+
+	var err error
+	// Before anything check to see that chain is not empty (for cold wallets)
```

Comment:
Isn't this checking if it is connected to daemon, not if chain is empty?
I guess that comment is for the whole code block but this makes it unclear.  Maybe a blank like then a comment above the first check.  Not sure what you'd want.